elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -309,21 +309,19 @@ export class PeerPool extends EventEmitter {
 	}
 
 	public async request(packet: P2PRequestPacket): Promise<P2PResponsePacket> {
-		const listOfPeerInfo = [...this._peerMap.values()].map(
-			(peer: Peer) => peer.peerInfo as P2PDiscoveredPeerInfo,
-		);
+		const listOfPeerInfo = [...this._peerMap.values()]
+			.filter(p => p.state === ConnectionState.OPEN)
```

Comment:
I don't think we need this filtering since:

- We delete the peer from our peerMap as soon as we detect that they've disconnected.
- Even if the peer state is connected, it still does not fully guarantee that the peer is actually connected right now because WebSocket/TCP relies on a ping/pong mechanism to detect unintentional lost connection so there can be a delay between the lost connection and when it is detected (and the socket/peer state is updated).